File: 24. Swap Nodes in Pairs/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    def swapPairs(self, linklist):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        head = ListNode(0)
        head.next = linklist
        p = head
        while p.next is not None:
            temp = p.next.next
            if temp is None:
                return head.next
            p.next.next = temp.next
            temp.next = p.next
            p.next = temp
            p = temp.next
        return head.next

File: 24. Swap Nodes in Pairs/Solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution
{
  public:
    ListNode *swapPairs(ListNode *linklist)
    {
        ListNode *head = new ListNode(0);
        head->next = linklist;
        ListNode *p = head;
        while (p->next)
        {
            ListNode *temp = p->next->next;
            if (!temp)
                return head->next;
            p->next->next = temp->next;
            temp->next = p->next;
            p->next = temp;
            p = temp->next;
        }
        return head->next;
    }
};